src/anm/events.js

```diff
@@ -152,26 +152,28 @@ MouseEventsSupport.prototype.adaptEvent = function(event) {
 MouseEventsSupport.prototype.dispatch = function(event) {
     var owner = this.owner;
     var localEvent = this.adaptEvent(event);
-    var dispatchedByChild; // found the matching child inside
+    var dispatchedChild;  // found the matching child inside
     if (owner.inside(localEvent)) {
         owner.reverseEach(function(child) {
             if (child.isActive()) {
-                dispatchedByChild = child.dispatchMouseEvent(localEvent);
-                if (dispatchedByChild) return false; // stop iteration of reverseEach
+                dispatchedChild = child.dispatchMouseEvent(localEvent) ? child : null;
+                if (dispatchedChild) return false; // stop iteration of reverseEach
             }
         });
 
-        if (dispatchedByChild) return true;
-
-        if (event.type === 'mouseclick') {
-            this.owner.fire('mouseclick', localEvent);
-            return true;
-        } else if (event.type === 'mousemove') {
-            this.processMove(localEvent); // fire mouseenter/mousexit if required
-            return true;
+        if (!dispatchedChild && (event.type === 'mousemove')) {
+            this.processMove(localEvent); // fire mouseenter/mouseexit if required
         }
+
+        //var adaptedEvent = localEvent.clone();
+        //adaptedEvent.target = dispatchedChild || this.owner;
+        //this.owner.fire(event.type, adaptedEvent);
+
+        this.owner.fire(event.type, localEvent);
```

Comment:
it should be fired ONLY if none of the children handled it, i.e. 
```
   if (!dispatchedChild) {
       this.owner.fire(event.type, localEvent);
   }
```

also Anton, please have the tests to be exactly the same as in Kirill's tests. if they fail it means your implementation is incorrect and you need to fix the implementation, not change the tests!